main/src/modules/jointsAwareness/src/jointsAwarness.cpp

```diff
@@ -0,0 +1,413 @@
+{
+    moduleName = rf.check("name", Value("jointsAwareness")).asString().c_str();
+    setName(moduleName.c_str());
+
+    yInfo() << "findFileByName " << rf.findFileByName("jointsAwareness.ini") ;
+
+    cout << moduleName << ": finding configuration files..." << endl;
+    period = rf.check("period", Value(0.1)).asDouble();
+
+    bool    bEveryThingisGood = true;
+
+    robot   = rf.check("robot", Value("icub")).asString().c_str();
+    arm   = rf.check("arm", Value("both")).asString().c_str();
+    camera   = rf.check("camera", Value("left")).asString().c_str();
+
+    bEveryThingisGood &= configMaps();
+
+    string cameraSuffix ;
+    if(camera == "left" || camera == "right"){
+        cameraSuffix = "to";
+        cameraSuffix += camera;
+        cameraSuffix += "Cam";
+    } else {
+        yError() << "The camera used (" << camera << ") is NOT a valid one (left or right only)! Closing jointsAwareness!" ;
+        return false ;
+    }
+
+    if(arm == "left_arm" || arm == "right_arm"){
+        bEveryThingisGood &= configCartesian(arm, cameraSuffix);
+    } else if (arm == "both"){
+
+        yDebug() << "Both arm are used!" ;
+        bEveryThingisGood &= configCartesian("left_arm", cameraSuffix);
+        bEveryThingisGood &= configCartesian("right_arm", cameraSuffix);
+    } else {
+        yError() << "The arm used (" << arm << ") is NOT a valid one! Closing jointsAwareness!" ;
+        return false ;
+    }
+
+    string baseTorsoPortName = "/" + moduleName + "/" + "torso" + "/";
+    locPorts_map.find("torso")->second->open((baseTorsoPortName + "jointsLoc:o").c_str());
+    projPorts_map.find("torso")->second->open((baseTorsoPortName + cameraSuffix +"/joints2DProj:o").c_str());
+
+    Property optionGaze("(device gazecontrollerclient)");
+    optionGaze.put("remote","/iKinGazeCtrl");
+    optionGaze.put("local",("/"+moduleName+"/gaze").c_str());
+    if (!gazeClientCtrl.open(optionGaze))
+        yWarning("Gaze controller not available!");
+    if(gazeClientCtrl.isValid()){
+        gazeClientCtrl.view(iGaze);
+    } else {
+        yError() << "Invalid PolyDriver iGaze: exit!";
+        return false;
+    }
+
+    //rpc port
+    string read_ObjLoc_PortName = "/" + moduleName + "/" + "objects" + "/ojectsLoc:i";
+    read_ObjLoc_Port.open(read_ObjLoc_PortName);
+
+    string objLocStreamPortName = "/iol2opc/objLoc:o";
+
+    unsigned int counter = 0;
+    bool isConnected = false;
+    while(!isConnected && counter < 3){
+        yInfo() << "Trying to connect to " << read_ObjLoc_PortName ;
+        isConnected = Network::connect(objLocStreamPortName, read_ObjLoc_PortName);
+        counter++;
+        Time::delay(1.0);
+    }
+    if(!isConnected){
+               yWarning() << "The port " << objLocStreamPortName << " from iol2opc is needed to stream the 2D proj of objects! jointsAwareness will NOT be able to stream this inforations" ;
+               isObjectStreamed = false;
```

Comment:
This variable is unnecessary. You can use `read_ObjLoc_Port.getInputCount() > 0` instead.